cloudimg/common.py

```diff
@@ -60,4 +52,24 @@ class BaseService(object):
     """
     Base class for all cloud provider services.
     """
-    pass
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def publish(self, metadata):
+        """
+        Upload a VM image into a cloud marketplace and make it available for
+        all required accounts/groups.
+
+        Args:
+            metadata (PublishingMetadata): metadata for the VM image.
+        Returns:
+            The published object depending on the cloud.
+        """
+        raise NotImplementedError
+
+    def upload(self, metadata):
```

Comment:
What is the reason for having an alias?

I'd suggest not adding it because it seems to invite someone to inherit from this class and then implement 'publish' and 'upload' to do two different things, thus breaking the contract that it's just an alias.

Reply:
Agreed, I removed it.